src/modm/board/nucleo_f746zg/board.hpp

```diff
@@ -0,0 +1,144 @@
+/*
+ * Copyright (c) 2016-2017, Sascha Schade
+ * Copyright (c) 2016-2018, Niklas Hauser
+ *
+ * This file is part of the modm project.
+ *
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+// ----------------------------------------------------------------------------
+
+#ifndef MODM_STM32_NUCLEO_F746ZG_HPP
+#define MODM_STM32_NUCLEO_F746ZG_HPP
+
+#include <modm/platform.hpp>
+#include <modm/architecture/interface/clock.hpp>
+#include <modm/debug/logger.hpp>
+/// @ingroup modm_board_nucleo_f746zg
+#define MODM_BOARD_HAS_LOGGER
+
+using namespace modm::platform;
+
+/// @ingroup modm_board_nucleo_f746zg
+namespace Board
+{
+	using namespace modm::literals;
+
+/// STM32F7 running at 216MHz from the external 25MHz clock
+/// STM32F7 running at 216MHz from the external 25MHz clock
+struct SystemClock
+{
+	static constexpr uint32_t Frequency = 216_MHz;
+	static constexpr uint32_t Apb1 = Frequency / 8;
+	static constexpr uint32_t Apb2 = Frequency / 4;
```

Comment:
See below.

Reply:
I understand that the settings are wrong. I am very sorry. So the input clock is 8Mhz. But I have no clue, how Apb1/2 have to be set. I would appreciate a hint very much.

Reply:
No problem.
I will add a suggestion of changes (with explanations) later.

Reply:
## How to create the `SystemClock` struct
The easies way using STs CubeMX tool.

### 1. CubeMX clock graph

First we create a project in CubeMX with the desired microcontroller using the largest (pin-count, flash) variant.
CubeMX will display something like this in the "Clock configuration" tab:
![ST CubeMX Clock Configuration example](https://user-images.githubusercontent.com/2820734/81305132-43715000-907e-11ea-8b8e-b444dc7560cf.png)

Than configure all the clocks, muxes, multipliers and dividers to the highest allowed clock speeds (*).

(*) Some exceptions: USB mostly needs exactly 48 MHz, tbc...

This settings are reflected in the `static constexpr uint32_t Frequency`, `Apb1` and `Apb2` constants and in `const Rcc::PllFactors pllFactors{...}` and the following lines.
The `PllFactors` struct should be pretty self-explanatory.


### 2. Peripheral mapping to clocks

As we can see in the graphic above, there are different clock domains. Each peripheral is connected to a clock domain.
Some peripherals have an upstream clock mux, this is currently ignored in modm and the default clock mux setting is assumed.

The mapping is shown is the controllers block diagram located at the begin of the datasheet (not in the reference manual).

![grafik](https://user-images.githubusercontent.com/2820734/81448035-fffe0b00-917d-11ea-8a1f-428548851e63.png)

For every peripheral we create a `static constexpr uint32_t` member in the `struct SystemClock` and assign the value of the clock domain the peripheral is connected to.